### PR TITLE
@alloy => everything 3

### PR DESCRIPTION
**Pull request by orta**
Sorry dude, I wanted to do this last night, but I ended up too excited about @krausefx and couldn't write anymore code or do a PR to ash.

Majors:
- Artwork pagination for the show VC
- Lets you ask for a SMS with the Artsy app in it
- Fixes margins to be consistent with apple's recommendations
- Adds a tonne of caching for images
- A hack for apple's new arbitrary check for framework bundle ids, https://github.com/CocoaPods/CocoaPods/issues/4421

### CODE REVIEW

File: Emergence/Contexts/ArtworkViewController.swift [hunk 1]
```diff
@@ -21,8 +22,30 @@ class ArtworkViewController: UIViewController {
...
+
+            let height = artworkPreviewImage.bounds.height
+            guard let fullThumbnailPath = actualImage.bestThumbnailWithHeight(height) else { return print("no thumbnail for artwork") }
+
+            let manager = SDWebImageManager.sharedManager()
+            if let previousThumbnail = actualImage.bestThumbnailWithHeight(showArtworksHeight) where manager.cachedImageExistsForURL(previousThumbnail) {
+                let key = manager.cacheKeyForURL(previousThumbnail)
+                let smallerInitialThumbnail = manager.imageCache.imageFromMemoryCacheForKey(key)
+
+                artworkPreviewImage.sd_setImageWithURL(fullThumbnailPath, placeholderImage: smallerInitialThumbnail)
```
Comment (ashfurrow) [inline Emergence/Contexts/ArtworkViewController.swift:45]: Too bad SDWebImage doesn't have this built-in. Something like "load this image URL, but if you don't have it cached you can use this other image URL in the mean time."

Comment (orta) [inline Emergence/Contexts/ArtworkViewController.swift:45]: Yeah, I figured if I needed it again I'd make it an extension

Comment (ashfurrow) [inline Emergence/Contexts/ArtworkViewController.swift:45]: :ok_hand:

**Comment by ashfurrow** [comment]
Wonderful :100:

**Comment by KrauseFx** [comment]
:+1:

**Pull request merged**